src/modules/dirwatcher.js

```diff
@@ -0,0 +1,37 @@
+import EventEmitter from 'events'
+import fs from 'fs'
+import Set from 'collections/set'
+
+export default class DirWatcher extends EventEmitter {
+
+    constructor(path, delay) {
+        super();
+        this.allFiles = new Set();
+        this.watch(path, delay);
+    }
+
+    watch(path, delay) {
+        setInterval(() => {
+            fs.readdir(path, (error, files) => {
+                if (error) {
+                    console.log(error);
+                    throw error;
+                } else {
+                    if (this.allFiles.length == 0) {                        
```

Comment:
Don't use `==`, use only strict comparison `===`. Wonder why? Ask =)
And also in this case you can simply use `if (!this.allFiles.length)` as in JS 0 will be transformed to boolean `false` and after it'll look smth like `if (!false)`